docs/src/developers_guide/testing_style_guide.rst

```diff
@@ -0,0 +1,67 @@
+.. include:: ../common_links.inc
+
+.. _testing_style_guide:
+
+PyTest Style Guide
+******************
+
+This style guide should be approached pragmatically. Most of the guidelines laid out
+below will not be practical in every scenario, and as such should not be considered
+firm rules.
+
+At time of writing, some existing tests have already been written in PyTest,
+so might not be abiding by these guidelines.
+
+Directory
+=========
+
+Where suitable, tests should be located within the relevant directories.
+In most circumstance, that means new tests should not be placed in the
+root test directory, but in the relevant sub-folders.
+
+Conftest.py
+===========
+
+There should be a conftest.py file in the root/unit and root/integration
+folders. Additional lower level conftests can be added if it is agreed there
+is a need.
+
+Fixtures
+========
+
+As far as is possible, the actual test function should do little else but the
+actual assertion. However, in some cases this will not be applicable, so this
+will have to be decided on a case by case basis.
```

Comment:
I'd prefer less vague phrasing. You want to avoid stuff like:

>This is true except when it is not

You've done a great job of more specific phrasing in the following paragraph. So how to do the same with this one... The conflict here is readability versus maintainability:

- Max readability = the whole sequence written in a single test file - you can see everything in one place
- Max maintainability = all setup abstracted into fixtures - you can alter all relevant tests with a single change, distinction is made between setup and test when things go wrong

Something like the below?

```suggestion
actual assertion. Separating preparation into fixtures can make the code
harder to follow, so compromises are acceptable; e.g. setting up a test ``Cube``
should be a fixture, creating a simple string (``expected = "foo"``) for a single
test should not be a fixture. 
```